packages/source-map-utils/index.js

```diff
@@ -109,8 +109,10 @@ var SourceMapUtils = {
     }
 
     //because we might be dealing with a constructor with arguments, we do
-    //*not* remove metadata manually
-    let instructions = CodeUtils.parseCode(binary, numInstructions);
+    //*not* pass attemptStripMetadata under any circumstances
```

Comment:
I think it would be helpful to explain why metadata is preserved for potential cstor with artuments.

Reply:
...so, I was going to explain why, but on writing out the explanation, I realized this doesn't actually make sense anymore!  So I'll explain why I did it this way, and then I'll go change it. :P

Suppose you have a constructor, and you attempt to remove the metadata.  But there are arguments on the end of the constructor -- meaning it's the arguments on the end, not the metadata!  So the attempt to remove the metadata could go very badly.  More specifically, if the last two bytes get interpreted as a large length, then a whole lot could be incorrectly removed, way more than just metadata!

The reason why this no longer applies is because these days we only remove valid metadata.  The probability of us removing code by accident is quite low.  That said, if the last two bytes are *small*, we could notionally remove parts of the *arguments* by accident... but who cares, arguments are data and not code!

Hm, I dunno.  Maybe it's best to leave this as-is after all.  This case honestly doesn't really matter that much.  But I can at least explain it more?